fastn-core/src/db.rs

```diff
@@ -16,3 +16,25 @@ static POOL_RESULT: tokio::sync::OnceCell<fastn_core::Result<PgPool>> =
 pub async fn pool() -> &'static fastn_core::Result<PgPool> {
     POOL_RESULT.get_or_init(create_pool).await
 }
+
+/// run migrations on `db_url`
+pub fn migrate(db_url: String) -> fastn_core::Result<()> {
```

Comment:
Not too happy with it being sync operation. It can starve all our threads on `hostn`, and lead to DOS. We should convert it to async by using [cpupool](https://docs.rs/async-cpupool/latest/async_cpupool/).

Also there is this create: https://lib.rs/crates/diesel_async_migrations, means async migration should be "natively" possible with diesel.

Reply:
I have tried diesel_async_migrations (maybe 20-30 mins only), and gave up, trait issues in diesel + async etc are complete nightmare. So cpupool is good enough for now.

Reply:
well it worked first try for me https://github.com/fastn-stack/fastn/pull/1718/commits/0f2b3d7cd2fa626992d475bc8d6019b106f40150 :eyes: